pkg/virt-operator/creation/components/crds.go

```diff
@@ -272,6 +279,72 @@ func NewServiceMonitorCR(namespace string, monitorNamespace string, insecureSkip
 	}
 }
 
+// NewPrometheusRuleCR returns a PrometheusRule with a group of alerts for the KubeVirt deployment.
+func NewPrometheusRuleCR(namespace string) *promv1.PrometheusRule {
+	return &promv1.PrometheusRule{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: promv1.SchemeGroupVersion.String(),
+			Kind:       "PrometheusRule",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      KUBEVIRT_PROMETHEUS_RULE_NAME,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"prometheus.kubevirt.io": "",
+				"k8s-app":                "kubevirt",
+			},
+		},
+		Spec: *NewPrometheusRuleSpec(namespace),
+	}
+}
+
+// NewPrometheusRuleSpec makes a prometheus rule spec for kubevirt
+func NewPrometheusRuleSpec(ns string) *promv1.PrometheusRuleSpec {
+	return &promv1.PrometheusRuleSpec{
+		Groups: []promv1.RuleGroup{
+			{
+				Name: "kubevirt.rules",
+				Rules: []promv1.Rule{
+					{
+						Record: "num_of_running_virt_api_servers",
+						Expr: intstr.FromString(
+							fmt.Sprintf("sum(up{pod=~'virt-api-.+', namespace='%s'})", ns),
+						),
+					},
+					{
+						Alert: "VirtAPIDown",
+						Expr:  intstr.FromString("num_of_running_virt_api_servers == 0"),
+						For:   "5m",
+						Annotations: map[string]string{
+							"summary": "All VirtAPI servers are down.",
+						},
+					},
+					{
+						Alert: "SingleVirtAPI",
```

Comment:
Thinking about it `SingleVirtAPI` looks a little bit redundant compared to `LowVirtAPICount`. What do you think?